internal/model/todos.go

```diff
@@ -56,13 +59,15 @@ func CreateTodos(name string, todos []string) ([]Todo, error) {
 	sqlStatement := `INSERT INTO todos(list_id, todo) VALUES($1, $2)
 		RETURNING id, list_id, todo, done`
 	for _, todo := range todos {
-		var createdTodo = Todo{}
+		var createdTodo = &Todo{}
 		err := db.QueryRow(sqlStatement, listID, todo).Scan(&createdTodo.ID, &createdTodo.ListID, &createdTodo.Todo, &createdTodo.Done)
 		if err != nil {
 			return nil, err
-		} else if createdTodo.ID != "" {
-			newTodos = append(newTodos, createdTodo)
 		}
+		if createdTodo.ID == "" {
```

Comment:
So I switched this around a bit... instead of making the "right thing" nested in a n "else if" we check for the wrong thing and send an error up.

That way you can follow a straight line down to see what's happening and when something goes wrong we have a clear feedback loop (the error message).